metadata-ingestion/src/datahub/ingestion/reporting/datahub_ingestion_run_summary_provider.py

```diff
@@ -79,36 +79,33 @@ def create(
         cls,
         config_dict: Dict[str, Any],
         ctx: PipelineContext,
+        sink: Sink,
     ) -> PipelineRunListener:
-        sink_config_holder: Optional[DynamicTypedConfig] = None
-
         reporter_config = DatahubIngestionRunSummaryProviderConfig.parse_obj(
             config_dict or {}
         )
         if reporter_config.sink:
-            sink_config_holder = reporter_config.sink
-
-        if sink_config_holder is None:
-            # Populate sink from global recipe
-            assert ctx.pipeline_config
-            sink_config_holder = ctx.pipeline_config.sink
-            # Global instances are safe to use only if the types are datahub-rest and datahub-kafka
-            # Re-using a shared file sink will result in clobbering the events
-            if sink_config_holder.type not in ["datahub-rest", "datahub-kafka"]:
+            sink_class = sink_registry.get(reporter_config.sink.type)
+            sink_config = reporter_config.sink.config or {}
+            sink = sink_class.create(sink_config, ctx)
+        else:
+            if not isinstance(
+                sink,
+                tuple(
+                    [
+                        kls
+                        for kls in [
+                            sink_registry.get_optional("datahub-rest"),
+                            sink_registry.get_optional("datahub-kafka"),
+                        ]
+                        if kls
+                    ]
+                ),
+            ):
                 raise IgnorableError(
-                    f"Datahub ingestion reporter will be disabled because sink type {sink_config_holder.type} is not supported"
+                    f"Datahub ingestion reporter will be disabled because sink type {type(sink)} is not supported"
```

Comment:
It looks a bit weird we use a hardwired list for this, and not the sink has some property which can tell if it supports reporter or not

Reply:
yeah I don't really like the way this is set up either, but don't want to boil the ocean right now